API_doc/Communication.md

```diff
@@ -4,14 +4,20 @@
 
 All requests must contain **REQUEST_NAME**, containing the type of action they want to do.
 Each response contain at least **REQUEST_STATE** (see below).
+All the local settings are applied after the config inclusions.
 
 ### Request types
 | Request Name | Request Description | Additional Argument(s) | Additional Returned Value(s) |
 | -------- | -------- | -------- | -------- |
 |*CONFIG_CREATE*|Create a new config.|**CONFIG_NAME**|**CONFIG_KEY**<br>**READONLY_CONFIG_KEY**|
 |*CONFIG_LOAD*| Load an existing config. |**CONFIG_KEY** *or* **READONLY_CONFIG_KEY**| **CONFIG_NAME**<br>**CONFIG_ID**|
 |*CONFIG_UNLOAD*| Unload config. |**CONFIG_ID**|*none*|
-|*CONFIG_INCLUDE*| Include a config in another |**CONFIG_ID**<br>**SRC**<br>**DEST**|*none*|
+|*CONFIG_DELETE*| Delete config. |**CONFIG_ID**|*none*|
+|*CONFIG_GET_DEPS*| Get the ordered list of all included configs. |**CONFIG_ID**|**DEPS** (array of READONLY_CONFIG_KEY)|
+|*CONFIG_GET_SETTINGS*| Get the list of all local settings. |**CONFIG_ID**|**SETTINGS** (map of settings : "SETTING_NAME" -> "SETTING_VALUE")|
+|*CONFIG_GET_ALIASES*| Get the list of all local aliases. |**CONFIG_ID**|**ALIASES** (map of aliases : "ALIAS_NAME" -> "SETTING_NAME")|
+|*CONFIG_INCLUDE*| Include a config |**CONFIG_ID**<br>**SRC** (READONLY_CONFIG_KEY of config to include)<br>**INCLUDE_POSITION** (position in the list of inclusion, where 0 is the first to be included. If not specified, is equal to *0*)|*none*|
```

Comment:
I don't see the advantage of allowing the usage of the config_key. It just makes the implementation more verbose.

Reply:
I don't understand what you mean.

Reply:
how would you specify which configuration to include then if not using the config_key (or maybe a temporary id if you already loaded the config to include)?

Reply:
The service knows which config is behind a client's config_id. Passing the config_key seems superfluous, optimised and less user friendly when compared to passing a config_id. Plus, I believe config keys should be used as little as possible.

Reply:
in case I want to include a configI didn't load I can only do that bay passing the config_key, so in my opinion we should be able to pass either a key or temporary config_id

Reply:
Ok, but loading a config is easy, and can be hidden in the library. So I don't see the use of putting it in the communication protocol.

Reply:
So, if I understand well, you want that the user load the config before to include or to uninclude it ?

Reply:
The user could do this manually or call a library function which would do this implicitly on his behalf.
This is the communication API, not the client API.